src/support/borsh.rs

```diff
@@ -41,9 +41,25 @@ impl<const BITS: usize, const LIMBS: usize> BorshDeserialize for Uint<BITS, LIMB
 impl<const BITS: usize, const LIMBS: usize> BorshSerialize for Uint<BITS, LIMBS> {
     #[inline]
     fn serialize<W: io::Write>(&self, writer: &mut W) -> io::Result<()> {
-        // TODO: non-allocating and remove the `alloc` feature requirement
-        let bytes = self.as_le_bytes();
-        writer.write_all(&bytes[..Self::BYTES])
+        #[cfg(target_endian = "little")]
+        return writer.write_all(self.as_le_slice());
+
+        // TODO: Replace the unsafety with `generic_const_exprs` when
```

Comment:
do we have an issue for this filed?

Reply:
The action, but will auto file it when the PR is merged